src/main.rs

```diff
@@ -18,8 +19,15 @@ pub struct LinePart {
     tab_index: Option<usize>,
 }
 
+#[derive(Debug, Default, Clone)]
+struct WatchStatus {
```

Comment:
derive `Clone` only because the hashmap key/value needs it so that the whole hashmap can be cloned.
key is `usize`, already implements `Clone`.